dev/hal/src/DAC.cpp

```diff
@@ -0,0 +1,37 @@
+/**
+ * @file DAC.cpp
+ * @brief noura36
+ * @version 0.1
+ * @date 2024-08-02
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include "Pin.h"
+#include "Gpio.h"
+#include "Systick.h"
+#include "Array.h"
+#include "DAC.h"
+
+using  namespace stm32::dev::mcal::pin;
+using  namespace stm32::dev::mcal::gpio;
+using namespace stm32::dev::hal::dac;
+
+void Dac::DAC_Init(const Array<Pin, 8>& dacPins, CLKSource clock) {
+    for (uint8_t i = 0; i < dacPins.Size(); i++) {
```

Comment:
- Use `Rcc::Enable()` to enable any needed peripherals in `init()`
- I think it will be better if we do the initialization inside the constructor
- Also use `STM32_ASSERT()` to check if pin is in correct mode or not e.g. `pin.IsAnalog()`, `pin.IsInput()`, ....